### PR TITLE
device: make sure we don't generate bogus PropertiesChanged signals

### PR DESCRIPTION
**Pull request by msekletar**

### CODE REVIEW

**Comment by rmetrich** [comment]
# Example of test scenario

1. Set up a system with 200 mount points on a "data" volume group

~~~
# vgcreate data /dev/sdb
# for i in $(seq 1 200); do lvcreate -n lv$i -L 10M data; mkfs.ext4 /dev/data/lv$i; echo "/dev/mapper/data-lv$i /data/lv$i ext4 defaults 0 0" >> /etc/fstab; mkdir -p /data/lv$i; done
# systemctl daemon-reload
# mount -a
~~~

2. Monitor systemd's "sendmsg" while performing 100 remounts of /data/lv1

In a terminal:
~~~
# timeout 1m strace -ttTvyy -o /tmp/systemd.strace -s 1024 -e sendmsg -p 1
~~~

In another terminal:
~~~
# for i in $(seq 1 100); do mount -o remount /data/lv1; sleep 0.1; done
~~~

3. Count the number of PropertiesChanged messages sent

~~~
# grep -c PropertiesChanged /tmp/systemd.strace
~~~

Actual results:

40600

With the patch:

200

File: src/core/unit.c [hunk 1]
```diff
@@ -327,6 +327,9 @@ int unit_set_description(Unit *u, const char *description) {
 
         assert(u);
 
+        if (streq_ptr(u->description, description))
+                return 0;
+
         r = free_and_strdup(&u->description, empty_to_null(description));
```
Comment (keszybz) [inline src/core/unit.c]: Does this make any difference? free_and_strdup() already does nothing and returns 0 if the strings are the same.

**Comment by keszybz** [comment]
(I set the flag because I'm 80% sure that the first patch is necessary.)

File: src/core/unit.c (hunk 1 above)
Comment (msekletar) [reply src/core/unit.c]: I've dropped the patch and tested again and it seems that it is indeed not needed. Good catch. Thanks!

**Comment by msekletar** [comment]
@keszybz I've dropped the unnecessary patch. PTAL.

**Pull request merged**